### PR TITLE
MB-8135 updateMTOShipment won't blank out addresses 

### PR DESCRIPTION
**Pull request by sandy-wright**
## Description

Recently, I had updated the Prime API YAML file to have a new structure for address fields in shipments so that they could each have an individual description. However, this introduced a bug to the `updateMTOShipment` endpoint wherein an address would have its values blanked out if it was not included in the request body. Fixing this involved updating the `AddressModel` function in `payload_to_model` to accurately identify blank addresses. I also added a test and caught another little bug in the meantime.

## Reviewer Notes

One other effect of the update to the Prime API's address structure is the return values. Instead of omitting the field if a null value was sent back from the server, now the response will look like:

```json
"secondaryDeliveryAddress": {
  "city": null,
  "postalCode": null,
  "state": null,
  "streetAddress1": null
},
```

There is no way to get around this if we want to have visible documentation for each of these fields. The question - is this acceptable? Is the documentation not worth the potential confusion with the response? I'd appreciate feedback on that!

## Setup

Populate your db and run your server:

```sh
make db_dev_e2e_populate server_run
```

Using the Prime API testing tool of your choice ([Prime API Client](https://github.com/transcom/prime_api_deliverable/wiki/Prime-API-Client) or [Postman](https://github.com/transcom/prime_api_deliverable/wiki/Using-Postman)):

* Run `fetchMTOUpdates` to grab a shipment's ID and eTag 
* Save a record of one of its addresses (`pickupAddress`, `destinationAddress`, `secondaryPickupAddress`, or `secondaryDeliveryAddress`) so that you can refer to it later
* Go to `updateMTOShipment`. Fill in the ID and eTag in the path params and `IfMatch` header, respectively.
* Set up a _small_ request body. Something like:
  ```json
  {
    "scheduledPickupDate": "2021-06-23"
  }
  ```
* Send the request.
* Check out your chosen address field in the response. Are all the values the same as they were before you sent the update? Then yay, that's a success!

## Code Review Verification Steps

* [x] Request review from a member of a different team.
* [x] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-8135) for this change.

### CODE REVIEW

File: pkg/services/mto_shipment/mto_shipment_updater.go [hunk 1]
```diff
@@ -280,7 +280,7 @@ func (f *mtoShipmentUpdater) updateShipmentRecord(dbShipment *models.MTOShipment
 
 		if newShipment.SecondaryPickupAddress != nil {
 			if dbShipment.SecondaryPickupAddressID != nil {
-				newShipment.PickupAddress.ID = *dbShipment.SecondaryPickupAddressID
+				newShipment.SecondaryPickupAddress.ID = *dbShipment.SecondaryPickupAddressID
```
Comment (felipe-lee) [inline pkg/services/mto_shipment/mto_shipment_updater.go:283]: @sandy-wright This is the test we modified in @mlm55 's pr to start seeing if the addresses are properly being updated https://github.com/transcom/mymove/blob/6d8765a8b4266dfa752b9d569ff1928333cf7d38/pkg/services/mto_shipment/mto_shipment_updater_test.go#L151-L177

Comment (felipe-lee) [reply pkg/services/mto_shipment/mto_shipment_updater.go:283]: Specifically this:
https://github.com/transcom/mymove/pull/6608/files/813a6ebbcafebd21299d30e733db2c0e380124b2..c1f1391b2dfa798a988b405caafb869e2853895f

Just to try seeing some changes come through.

**Review by abbyoung — APPROVED** [review]
lgtm! thank you for catching and fixing

File: pkg/services/mto_shipment/mto_shipment_updater.go (hunk 1 above)
Comment (pearl-truss) [reply pkg/services/mto_shipment/mto_shipment_updater.go:283]: I pulled down this PR and added those test and I see that they're passing 🥳

Comment (sandy-wright) [reply pkg/services/mto_shipment/mto_shipment_updater.go:283]: woo, thank you!! Should I add those tests to mine, or just let your PR take care of that?

Comment (sandy-wright) [reply pkg/services/mto_shipment/mto_shipment_updater.go:283]: ...mm I'll actually go ahead and merge. I got them to pass too, and I would just be duplicating the work in that PR

**Pull request merged**